pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -124,6 +143,33 @@ func (c controller) List(typ config.GroupVersionKind, namespace string) ([]confi
 	return nil, errUnsupportedOp
 }
 
+func (r *KubernetesResources) UpdateStatuses(c controller) {
+	c.handleStatusUpdates(r.GatewayClass)
+	c.handleStatusUpdates(r.Gateway)
+	c.handleStatusUpdates(r.HTTPRoute)
+	c.handleStatusUpdates(r.TCPRoute)
+	c.handleStatusUpdates(r.TLSRoute)
+	c.handleStatusUpdates(r.BackendPolicy)
+}
+
+func (c controller) handleStatusUpdates(configs []config.Config) {
+	for _, cfg := range configs {
+		ws := cfg.Status.(*kstatus.WrappedStatus)
+
+		if ws.Dirty {
+			_, err := c.cache.UpdateStatus(config.Config{
+				Meta:   cfg.Meta,
+				Status: ws.Unwrap(),
+			})
+			// TODO make this more resilient. When we add a queue we can make transient failures retry
+			// and drop permanent failures
+			if err != nil {
+				log.Errorf("failed to update status for %v/%v: %v", cfg.GroupVersionKind, cfg.Name, err)
```

Comment:
I'm starting to add a boilerplate comment for all new 'log.Errorf': 

"Can you add a metric, so we can alert when this happens ?"